src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/Lightup/LightupCompilationWithAnalyzers.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Testing.Lightup
+{
+    /// <summary>
+    /// With Roslyn 4.x diagnostics that have been suppressed by a DiagnosticSuppressor are no longer reported when 'report suppressed diagnostics' is not explicitly enabled via options.
+    /// Since reporting suppression via DiagnosticSuppressor is now configured the same as reporting suppression via #pragma, 'report suppressed diagnostics' must be enabled during the diagnostic
+    /// test stage to be able to test a DiagnosticSuppressor, but must be disabled during the suppressor test stage.
+    /// To minimize the risk of side effects, 'report suppressed diagnostics' will only be enabled when testing a DiagnosticSuppressor.
+    /// </summary>
+    internal static class LightupCompilationWithAnalyzers
```

Comment:
📝 This would likely be merged into `CompilationWithAnalyzersExtensions` eventually